### PR TITLE
implements #1295

### PR DESCRIPTION
**Pull request by sbromberger**
cc @abhinavmehndiratta

### CODE REVIEW

File: src/SimpleGraphs/simpledigraph.jl [hunk 1]
```diff
@@ -566,3 +566,32 @@ function rem_vertices!(g::SimpleDiGraph{T},
 
     return reverse_vmap
 end
+
+function all_neighbors(g::SimpleDiGraph{T}, u::T) where T
+   common = Vector{T}()
```
Comment (abhinavmehndiratta) [inline src/SimpleGraphs/simpledigraph.jl]: I named the variable `common` in a hurry, could you change it to something more suitable ? Like `union_in_out` or something else ?

**Comment by simonschoelly** [comment]
Would it maybe make more sense, to implement `all_neighbors` as an iterator, so that we don't have so many allocations? If someone wants a vector, they could still call `collect` on that iterator.

**Comment by sbromberger** [comment]
> Would it maybe make more sense, to implement all_neighbors as an iterator, so that we don't have so many allocations? If someone wants a vector, they could still call collect on that iterator.

Fine by me, but I'm worried about the inconsistencies between the various neighbor functions now.

**Comment by abhinavmehndiratta** [comment]
I think you need to import the function before overriding. I'm not sure though, maybe something related to trait functions is going on.

**Comment by sbromberger** [comment]
@simonschoelly I'm inclined to merge this given the immediate performance improvements. If you want to discuss iteration over `*neighbors`, can we do so in a GitHub issue?

**Pull request merged**